### PR TITLE
feat: ci storage reports

### PR DESCRIPTION
**Pull request by 0xClandestine**
Adapted from @stevennevins's [implementation](https://github.com/stevennevins/ci-workflows/blob/main/.github/workflows/storage-checker.yml)

### CODE REVIEW

File: .github/workflows/storage-report.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: Check Storage Layout
+on:
+  workflow_dispatch:
+  pull_request:
```
Comment (gpsanant) [inline .github/workflows/storage-report.yml:4]: can we do this on pull request into dev

File: .github/workflows/storage-report.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        run: |
+          git fetch origin $MAIN
+          git checkout $MAIN
+
+      - name: "Generate and prepare the contract artifacts for main branch"
+        run: |
+          bash bin/storage-report.sh
+
+      - name: Compare outputs
+        run: |
```
Comment (stevennevins) [inline .github/workflows/storage-report.yml:36]: In my workflow I was outputting to two separate directories and the diffing the contents of the two directories

Comment (0xClandestine) [reply .github/workflows/storage-report.yml:36]: Nice catch, will fix.

File: .github/workflows/storage-report.yml (hunk 1 above)
Comment (0xClandestine) [reply .github/workflows/storage-report.yml:4]: fixed

**Review by gpsanant — APPROVED** [review]
let's try it out

**Pull request merged**